src/test/java/com/spotify/futures/CompletableFuturesTest.java

```diff
@@ -51,12 +60,30 @@
 import static org.hamcrest.core.Is.isA;
 import static org.hamcrest.core.IsNot.not;
 import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 public class CompletableFuturesTest {
 
+  private ScheduledExecutorService executor ;
+
   @Rule
   public ExpectedException exception = ExpectedException.none();
 
+  @Before
+  public void setUp() {
+    executor = Executors.newScheduledThreadPool(1);
```

Comment:
Can we use something like this in the tests instead?

http://www.jmock.org/javadoc/jmock-2.6.0/doc/org/jmock/lib/concurrent/DeterministicScheduler.html